src/webapp/hooks/useAppConfig.ts

```diff
@@ -0,0 +1,60 @@
+import { useAppContext } from "../contexts/app-context";
+import { defaultConfig, PersistedConfig } from "../../data/entities/PersistedConfig";
+import React, { useMemo, useState } from "react";
+import { getCustomizableAppText } from "../../domain/entities/CustomText";
+import _ from "lodash";
+
+export function useAppConfig() {
+    const { usecases } = useAppContext();
+    const [appConfig, setAppConfig] = React.useState<PersistedConfig>(defaultConfig);
+    const [hasSettingsAccess, setHasSettingsAccess] = useState(false);
+    const [hasLoaded, setHasLoaded] = useState(false);
+    const appCustomText = useMemo(() => getCustomizableAppText(appConfig?.customText || {}), [appConfig]);
+    const logoInfo = useMemo(() => getLogoInfo(appConfig?.logo), [appConfig]);
+
+    const save = React.useCallback(
+        async (config: PersistedConfig) => {
+            return usecases.config.save(config).then(setAppConfig);
```

Comment:
Shouldn't we use `await`?

Reply:
return promise and return await promise are equivalent (except for error handling, not relevant for this snippet), so we can keep it as-is. We could say "async" is not strictly needed, but nothing wrong with it.